server/app.js

```diff
@@ -3,30 +3,52 @@ const express = require("express");
 const { join } = require("path");
 const cookieParser = require("cookie-parser");
 const logger = require("morgan");
-
-const indexRouter = require("./routes/index");
-const pingRouter = require("./routes/ping");
-const authRouter = require("./routes/auth");
-const usersRouter = require("./routes/users");
-const chatRouter = require("./routes/chat");
-const MessageRouter = require("./routes/messages");
+const cors = require("cors");
 
 const { json, urlencoded } = express;
 
 const app = express();
 
+const corsOptions = {
+	origin: ["http://localhost:3000"],
+	methods: ["GET", "POST", "PATCH"],
+};
+
+app.use(cors(corsOptions));
 app.use(logger("dev"));
 app.use(json());
 app.use(urlencoded({ extended: false }));
 app.use(cookieParser());
 app.use(express.static(join(__dirname, "public")));
 
+const io = require("socket.io")({ cors: corsOptions });
+app.io = io;
+
+const registerSocketAuth = require("./helpers/socket/auth");
+const registerSocketChatRooms = require("./helpers/socket/chatRooms");
+
+//functions with registered event handlers
+const onConnection = (socket) => {
+	registerSocketAuth(io, socket);
+	registerSocketChatRooms(io, socket);
+};
+
+const { socketAuth } = require("./middleware/auth");
+
+io.use(socketAuth);
+io.on("connection", onConnection);
```

Comment:
I'd recommend having your socket initialization code in a separate file.

Reply:
will refactor